File: LeetCode_Solutions/LeetCode_Solutions/KeyboardRow_500.m
//
//  KeyboardRow_500.m
//  LeetCode_Solutions
//
//  Created by Huni on 23/03/2017.
//  Copyright © 2017 KnightJoker. All rights reserved.
//



#import "KeyboardRow_500.h"

NSString *const kkeyBoardRowOne = @"qwertyuiop";
NSString *const kkeyBoardRowTwo = @"asdfghjkl";
NSString *const kkeyBoardRowThree = @"zxcvbnm";

@implementation KeyboardRow_500

- (NSArray <NSString *>*)findWords:(NSArray <NSString *>*)words {

    
    NSMutableArray *resultArray = [[NSMutableArray alloc] init];
    
    
    for (NSString *word in words) {
        for(int i = 0; i < word.length; i++){
            UniChar ch = [word characterAtIndex:i];
            
        }
    }
    
    
    
    return nil;
}

@end

File: LeetCode_Swift/LeetCode_Swift/KeyboardRow_500.swift
//
//  KeyboardRow_500.swift
//  LeetCode_Swift
//
//  Created by Huni on 30/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

class KeyboardRow_500 {
    func findWords(_ words: [String]) -> [String] {
        
        let rules =  ["QWERTYUIOP", "ASDFGHJKL", "ZXCVBNM"]
        var result = words
        
        words.forEach { word in
            if (checkWords(word: word.uppercased())) {
                result = result.filter { $0 != word }
            }
        }

        func checkWords(word: String) -> Bool {
            
            let rule = rules.filter { $0.range(of: String(word.first!)) != nil }
            for c in word {
                if (rule.first!.range(of: String(c))) == nil {
                    return true
                }
            }
            
            return false
        }
        
        return result
    }
}
